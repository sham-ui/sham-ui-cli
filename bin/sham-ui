#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .parse( process.argv );