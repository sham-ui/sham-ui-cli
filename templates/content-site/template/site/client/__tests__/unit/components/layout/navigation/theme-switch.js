import { createDI } from 'sham-ui';
import * as directives from 'sham-ui-directives';
import renderer from 'sham-ui-test-helpers';
//eslint-disable-next-line max-len
import LayoutNavigationThemeSwitch  from '../../../../../src/components/layout/navigation/theme-switch.sfc';

it( 'renders correctly', () => {
    const meta = renderer( LayoutNavigationThemeSwitch, {}, {
        directives
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'toggle theme', () => {
    const DI = createDI();
    const toggle = jest.fn();
    DI.bind( 'dark-theme', {
        toggle: toggle
    } );
    const meta = renderer( LayoutNavigationThemeSwitch, {}, {
        DI,
        directives
    } );
    meta.ctx.container.querySelector( '#checkbox' ).click();
    expect( meta.toJSON() ).toMatchSnapshot();
    expect( toggle ).toHaveBeenCalledTimes( 1 );
} );
