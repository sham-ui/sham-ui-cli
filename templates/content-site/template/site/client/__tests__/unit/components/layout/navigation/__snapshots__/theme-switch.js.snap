// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "appState": Object {
      "darkThemeEnabled": false,
      "routerResolved": false,
    },
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "onChangeUpdate": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
  },
  "html": "
<div class=\\"theme-switch-wrapper\\">
  <label class=\\"theme-switch\\" for=\\"checkbox\\">
    <input type=\\"checkbox\\" id=\\"checkbox\\">
    <div class=\\"slider round\\"></div>
  </label>
</div>
",
}
`;

exports[`toggle theme 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "appState": Object {
      "darkThemeEnabled": false,
      "routerResolved": false,
    },
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "onChangeUpdate": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
  },
  "html": "
<div class=\\"theme-switch-wrapper\\">
  <label class=\\"theme-switch\\" for=\\"checkbox\\">
    <input type=\\"checkbox\\" id=\\"checkbox\\">
    <div class=\\"slider round\\"></div>
  </label>
</div>
",
}
`;
